pkg/lib/account_identifier.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package lib
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/moov-io/bai2/pkg/util"
+)
+
+const (
+	aiParseErrorFmt    = "AccountIdentifier: unable to parse %s"
+	aiValidateErrorFmt = "AccountIdentifierCurrent: invalid %s"
+)
+
+/*
+
+Account Identifier and Summary/Status for Current (CDA), Personal (PDA), and Loan accounts
+
+CDA and PDA accounts
+This record contains information on opening and closing balances for CDA and PDA accounts. It always
+has a record code of 03.
+
+*/
+
+// Creating CDA and PDA accounts
+func NewAccountIdentifier() *AccountIdentifier {
+	return &AccountIdentifier{
+		RecordCode: "03",
+	}
+}
+
+// CDA and PDA accounts
+type AccountIdentifier struct {
+	RecordCode    string
+	AccountNumber string
+	CurrencyCode  string   `json:",omitempty"`
+	TypeCode      string   `json:",omitempty"`
+	Amount        string   `json:",omitempty"`
+	ItemCount     string   `json:",omitempty"`
+	FundsType     string   `json:",omitempty"`
```

Comment:
So it turns out the Funds Type field is much more complicated than pretty much anything else in this file format (see Page 29 of the spec). It has a variable number of subfields, which confoundingly _also_ are comma-delimited. This significantly complicates scanning these fields by looking for the next comma. So we're going to need some dedicated logic to handle this, also keeping in mind that in Transaction Detail Records there are still fields (Bank Reference Number, Customer Reference Number, and Text) that can come after Funds Type. Let me know if you'd like any help on this.

Reply:
my question is how we can expect following fields by funds type